lib/bootloader_setup.pm

```diff
@@ -519,6 +521,33 @@ sub select_bootmenu_language {
     }
 }
 
+sub select_kernel {
+    my $kernel = shift;
+
+    assert_screen ['grub2', "grub2-$kernel-selected"], 100;
+    if (match_has_tag "grub2-$kernel-selected") {    # if requested kernel is selected continue
+        send_key 'ret';
+    }
+    else {                                           # else go to that kernel thru grub2 advanced options
+        send_key_until_needlematch 'grub2-advanced-options', 'down';
+        send_key 'ret';
+        send_key_until_needlematch "grub2-$kernel-selected", 'down';
+        send_key 'ret';
+    }
+    if (get_var('NOAUTOLOGIN')) {
+        my $ret = assert_screen 'displaymanager', 200;
+        mouse_hide();
+        if (get_var('DM_NEEDS_USERNAME')) {
```

Comment:
is possible that NOAUTOLOGIN is defined and DM_NEEDS_USERNAME is not ? 
For me it's looks over complicated we can reach same result using single variable.
Less code less bugs :)

Reply:
Well… This is code that @okurz just shifted from one file into another so it is not him to blame ;) Technically speaking something that has to do with the "displaymanager" does not belong in a function called "select_kernel" anyway.

Reply:
I moved that code to tests/rt/kmp_modules.pm, the only place where it's used. See the (old) code in action e.g. in https://openqa.suse.de/tests/1305377